include/openenclave/host.h

```diff
@@ -262,6 +262,154 @@ oe_result_t oe_verify_report(
+ */
+typedef enum _oe_asymmetric_key_format {
+    /**
+     * The PEM format.
+     */
+    OE_ASYMMETRIC_KEY_PEM = 1,
+
+    /**
+     * Unused.
+     */
+    _OE_ASYMMETRIC_KEY_FORMAT_MAX = OE_ENUM_MAX,
+} oe_asymmetric_key_format_t;
+/**< typedef enum _oe_asymmetric_key_format oe_asymmetric_key_format_t*/
+
+typedef struct _oe_asymmetric_key_params
+{
+    /**
+     *  The type of asymmetric key.
+     */
+    oe_asymmetric_key_type_t type;
+
+    /**
+     * The exported format of the key.
+     */
+    oe_asymmetric_key_format_t format;
+
+    /**
+     * Optional user data to add to the key derivation.
+     */
+    void* user_data;
+
+    /**
+     * The size of user_data.
+     */
+    size_t user_data_size;
+} oe_asymmetric_key_params_t;
+/**< typedef enum _oe_asymmetric_key_params oe_asymmetric_key_params_t*/
+
+/**
+ * Returns a public key that is associated with the identity of the enclave
+ * and the specified policy.
+ *
+ * @param enclave The enclave handle.
+ * @param seal_policy The policy for the identity properties used to derive
+ * the key.
+ * @param key_params The parameters for the asymmetric key derivation.
+ * @param key_buffer A pointer to the buffer that on success contains the
+ * requested public key.
+ * @param key_buffer_size On success, this contains size of key_buffer.
+ * @param key_info Optional pointer to a buffer for the enclave-specific key
+ * information which can be used to retrieve the same key later on a newer
+ * security version.
+ * @param key_info_size On success, this contains the size of key_info.
+ *
+ * @retval OE_OK The key was successfully requested.
+ * @retval OE_INVALID_PARAMETER At least one parameter is invalid.
+ * @retval OE_OUT_OF_MEMORY There is no memory available.
+ * @retval OE_UNEXPECTED An unexpected error happened.
+ */
+oe_result_t oe_get_public_key_by_policy(
+    oe_enclave_t* enclave,
+    oe_seal_policy_t seal_policy,
+    const oe_asymmetric_key_params_t* key_params,
+    uint8_t** key_buffer,
+    size_t* key_buffer_size,
+    uint8_t** key_info,
+    size_t* key_info_size);
+
+/**
+ * Returns a public key that is associated with the identity of the enclave.
+ *
+ * @param enclave The enclave handle.
+ * @param key_params The parameters for the asymmetric key derivation.
+ * @param key_info The enclave-specific key information to derive the key.
+ * @param key_info_size The size of the key_info buffer.
+ * @param key_buffer A pointer to the buffer that on success contains the
+ * requested public key.
+ * @param key_buffer_size On success, this contains size of key_buffer.
+ *
+ * @retval OE_OK The key was successfully requested.
+ * @retval OE_INVALID_PARAMETER At least one parameter is invalid.
+ * @retval OE_INVALID_CPUSVN The platform specific data has an invalid CPUSVN.
+ * @retval OE_INVALID_ISVSVN The platform specific data has an invalid ISVSVN.
+ * @retval OE_INVALID_KEYNAME The platform specific data has an invalid KEYNAME.
+ */
+oe_result_t oe_get_public_key(
```

Comment:
Does this default to one of the policies or use a completely different policy?

Reply:
The policy is in the `key_info` field. The `oe_get_X_key` functions have the `key_info` field as an "in" parameter. This is different from the `oe_get_X_key_by_policy` functions, where the user specifies the policy and the `key_info` is an "out" parameter. I personally think the naming of the two functions is confusing since they are actually more different than what their name implies, but that's how the original key API was written.